# HG changeset patch
# User Rik <rik@octave.org>
# Date 1467867308 25200
#      Wed Jul 06 21:55:08 2016 -0700
# Node ID 05ad2381015025bbd7055fdf3a5a12011383e234
# Parent  21f3cbd5c70825e320b730969ad2257f630beaa2
Use default gridlinestyle of '-' for Matlab compatibility.

* graphics.cc (axes::properties::set_defaults): Use '-' for default
gridlinestyle.

* graphics.in.h (AXES_PROPERTIES): Use '-' for default gridlinestyle.

* grid.m: Don't change gridlinestyle or minorgridlinestyle properties.

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -4850,17 +4850,17 @@ axes::properties::set_defaults (base_gra
   drawmode = "normal";
 
   fontangle = "normal";
   fontname = OCTAVE_DEFAULT_FONTNAME;
   fontsize = 10;
   fontunits = "points";
   fontweight = "normal";
 
-  gridlinestyle = ":";
+  gridlinestyle = "-";
   linestyleorder = "-";
   linewidth = 0.5;
   minorgridlinestyle = ":";
 
   // Note: plotboxaspectratio will be set through update_aspectratios
   plotboxaspectratiomode = "auto";
   projection = "orthographic";
 
diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -3835,17 +3835,17 @@ public:
       double_property fontsize u , 10
       radio_property fontunits SU , "{points}|inches|centimeters|normalized|pixels"
       bool_property fontsmoothing , "on"
       radio_property fontweight u , "{normal}|bold"
       double_property gridalpha m , 0.15
       radio_property gridalphamode , "{auto}|manual"
       color_property gridcolor , color_property (color_values (0.15, 0.15, 0.15), radio_values ("none"))
       radio_property gridcolormode , "{auto}|manual"
-      radio_property gridlinestyle , "{:}|-|--|-.|none"
+      radio_property gridlinestyle , "{-}|--|:|-.|none"
       double_property labelfontsizemultiplier , 1.1
       radio_property layer u , "{bottom}|top"
       // FIXME: should be kind of string array.
       any_property linestyleorder S , "-"
       double_property linestyleorderindex , 1.0
       double_property linewidth , 0.5
       double_property minorgridalpha m , 0.25
       radio_property minorgridalphamode , "{auto}|manual"
diff --git a/scripts/plot/appearance/grid.m b/scripts/plot/appearance/grid.m
--- a/scripts/plot/appearance/grid.m
+++ b/scripts/plot/appearance/grid.m
@@ -49,16 +49,17 @@
 function grid (varargin)
 
   [hax, varargin, nargs] = __plt_get_axis_arg__ ("grid", varargin{:});
 
   if (isempty (hax))
     hax = gca ();
   endif
 
+  ## Must be after gca (), since grid with no plot should create one.
   if (nargs > 2)
     print_usage ();
   endif
 
   grid_on = any (strcmp (get (hax, {"xgrid", "ygrid", "zgrid"}), "on"));
 
   minor_on = any (strcmp (get (hax, {"xminorgrid", "yminorgrid", "zminorgrid"}),
                          "on"));
@@ -94,21 +95,19 @@ function grid (varargin)
         endif
       endif
     else
       print_usage ();
     endif
   endif
 
   if (grid_on)
-    set (hax, "xgrid", "on", "ygrid", "on", "zgrid", "on",
-              "gridlinestyle", ":");
+    set (hax, "xgrid", "on", "ygrid", "on", "zgrid", "on");
     if (minor_on)
-      set (hax, "xminorgrid", "on", "yminorgrid", "on", "zminorgrid", "on",
-                "gridlinestyle", "-", "minorgridlinestyle", ":");
+      set (hax, "xminorgrid", "on", "yminorgrid", "on", "zminorgrid", "on");
       xg = ifelse (strcmp (get (hax, "xscale"), "log"), "off", "on");
       yg = ifelse (strcmp (get (hax, "yscale"), "log"), "off", "on");
       zg = ifelse (strcmp (get (hax, "zscale"), "log"), "off", "on");
       set (hax, "xgrid", xg, "ygrid", yg, "zgrid", zg);
     elseif (minor_auto)
       xmg = ifelse (strcmp (get (hax, "xscale"), "log"), "on", "off");
       ymg = ifelse (strcmp (get (hax, "yscale"), "log"), "on", "off");
       zmg = ifelse (strcmp (get (hax, "zscale"), "log"), "on", "off");
